sample-notificator-domain/src/main/groovy/life/qbic/business/notification/create/CreateNotification.groovy

```diff
@@ -15,156 +16,89 @@ import life.qbic.business.subscription.fetch.FetchSubscriberOutput
-     * @param previousProjectCode ProjectCode of the previous Sample in a SampleCode list
-     * @return true if ProjectCode if equal, false otherwise
-     */
-    private static boolean projectCodeIsSame(String currentProjectCode, String previousProjectCode) {
-        return currentProjectCode == previousProjectCode
-    }
+    private void addNotificationForProject(Project project) {
 
-    /**
-     * Method extracting the ProjectCode from the SampleCode
-     * @param sampleCode SampleCode, which contains the ProjectCode
-     * @return projectCode of the provided SampleCode.
-     */
-    private static String getProjectCodeFromSample(String sampleCode) {
-        Integer projectCodeStartIndex = sampleCode.findIndexOf {"Q"}
-        String projectCode = sampleCode.substring(projectCodeStartIndex, 5)
-        return projectCode
-    }
-    /**
-     * Transfers the generated list of subscribers to the implementing class
-     * @param subscribers the retrieved list of subscribers with modified subscriptions
-     * @since 1.0.0
-     */
+        int failedQCCount = filterSamplesByStatus(project.sampleCodes, "SAMPLE_QC_FAIL").size()
+        int availableDataCount = filterSamplesByStatus(project.sampleCodes, "DATA_AVAILABLE").size()
 
-    @Override
-    void fetchedSubscribers(List<Subscriber> subscribers) {
-        try {
-            //The output of the fetchSubscriber Use case will be stored in the notificationPerSubscriber Map and used in the CreateNotification method
-            this.notificationPerSubscriber = createNotificationPerSubscriber(subscribers)
-        } catch(Exception e) {
-            output.failNotification("An error occurred during notification creation")
+        //4. get subscribers of this projects
+        List<Subscriber> subscribers = fetchSubscriberDataSource.getSubscriberForProject(project.code)
+
+        for (Subscriber subscriber : subscribers) {
+            notifications << new NotificationContent.Builder(subscriber.firstName, subscriber.lastName, subscriber.email,
+                    project.title, project.code, failedQCCount, availableDataCount).build()
         }
+
     }
 
-    /**
-     * Sends failure notifications that have been
-     * recorded during the use case.
-     * @param notification containing a failure message
-     * @since 1.0.0
-     */
-    @Override
-    void failNotification(String notification) {
-        output.failNotification(notification)
+    private List<String> filterSamplesByStatus(List<String> samples, String statusName) {
+
+        Predicate isOfStatus = code -> updatedSamplesWithStatus.get(code).toString() == statusName
+        List<String> filteredCodes = samples.stream().filter(isOfStatus).collect().toList()
+
+        return filteredCodes
+    }
+
+
+    private Set<Project> getProjects() {
+        Set<Project> projects = new HashSet<>()
+
+        updatedSamplesWithStatus.keySet().each { sample ->
```

Comment:
```suggestion
        updatedSamplesWithStatus.keySet().each { sampleCode ->
```
Maybe it would be an idea to rename the variable to make it clear that we only have the sample code and not another representation of the sample here.